src/index.ts

```diff
@@ -105,30 +103,45 @@ export class Tunnel extends EventEmitter {
 				.then(() =>
 					this.emit('connect', hostname, port, head)
 				);
-			}).catch((err) => {
+			}).catch((err: Error) => {
 				this.emit('error', err);
 				cltSocket.destroy();
 			})
 		);
 	}
 
+	use(middleware: Middleware) {
+		this.stack.push(middleware);
+	}
+
+	private handleMiddleware(req: Request, cltSocket: net.Socket, head: Buffer): Promise<void> {
+		return Promise.try(() => {
```

Comment:
Why the `Promise.try`?  If you want for all the middleware to apply then you probably want to use `new Promise` or `Promise.fromCallback` and resolve when `middleware == null`